### PR TITLE
fix #350, run and eval drops into shell with the last scope

### LINKED ISSUE #350: Drop into arrai shell on error
When a program encounters an error, fire up the interactive shell and make the current scope available as a global `@locals` tuple.

### PR DESCRIPTION
**Pull request by nofun97**
Fixes #350 .

Changes proposed in this pull request:
- `arrai run` and `arrai eval` will drop into the interactive shell with the last scope available to it

Checklist:
- [x] Added related tests
- [x] Made corresponding changes to the documentation

### CODE REVIEW

File: docs/tutorial/shell.md [hunk 1]
```diff
@@ -35,6 +35,32 @@ $ _
 
 In Windows, TODO.
 
+## Shell as debugger
+
+When you evaluate an arrai script by using and the script fails, the `arrai`
```
Comment (marcelocantos) [inline docs/tutorial/shell.md:40]: typo?

File: docs/tutorial/shell.md [hunk 2]
```diff
@@ -35,6 +35,32 @@ $ _
...
+program will drop into the `arrai` interactive shell with the scope near the
+point of failure available to the interactive shell as a tuple.
+
+This can be accessed through the `@locals` variable.
```
Comment (marcelocantos) [inline docs/tutorial/shell.md:44]: It's not obvious that you're referring to the scope.

File: docs/tutorial/shell.md [hunk 3]
```diff
@@ -35,6 +35,32 @@ $ _
...
+
+```bash
+❯ arrai e 'let x = 1; (\a a + b)(x)'
```
Comment (marcelocantos) [inline docs/tutorial/shell.md:47]: Use `$` for consistency.

File: rel/expr.go [hunk 4]
```diff
@@ -32,21 +32,50 @@ func (e ExprScanner) Source() parser.Scanner {
...
+		if currentErr, isContextErr = ctxErr.err.(ContextErr); !isContextErr {
+			return ctxErr.scope
+		}
+		ctxErr = currentErr
+	}
+}
+
+func wrapContext(err error, expr Expr, scope Scope) error {
+	// removes stdlib
+	scope = scope.Without(".")
```
Comment (marcelocantos) [inline rel/expr.go:60]: `.` isn't stdlib.

File: rel/expr.go [hunk 5]
```diff
@@ -32,21 +32,50 @@ func (e ExprScanner) Source() parser.Scanner {
...
+	if _, isContextErr := err.(ContextErr); isContextErr {
+		return ContextErr{err, expr.Source(), EmptyScope}
```
Comment (marcelocantos) [inline rel/expr.go:62]: Does it need to be removed? Preserving the chain of scopes might help when implementing stack frame management in the debugger (show me the local names in the function that called me).

File: rel/expr.go (hunk 4 above)
Comment (nofun97) [reply rel/expr.go]: removed

**Pull request merged**